mantidimaging/core/io/saver.py

```diff
@@ -91,11 +93,25 @@ def save(images: Images,
     output_dir = os.path.abspath(os.path.expanduser(output_dir))
     make_dirs_if_needed(output_dir, overwrite_all)
 
+    # Do rescale if needed.
+    if pixel_depth is None or pixel_depth == "float32":
+        rescale_params = None
+    elif pixel_depth == "int16":
+        int16_size = 65536
+        max_value = images.data.max()
+        slope = max_value / int16_size
+        rescale_params = {"offset": 0, "slope": slope}
```

Comment:
Note that even with moving the rescaling to the for-loop, these parameters still need to be recorded in the metadata.

Reply:
:+1: